text/0027-disabling-bids.md

```diff
@@ -39,7 +39,7 @@ In either case, we would want to disable such a validator's bid. This would be d
 
 The validity of this transaction would be determined by the consensus state as seen by the switch block proposal consensus unit (i.e. the number of messages sent by each validator would be calculated based on what the unit containing the switch block sees) - this ensures that all validators will agree about whose bids need to be disabled.
 
-The new transaction would completely disable the validator's bid upon execution by the auction contract and remove the offending validator from the validator set from the next era onwards. The funds would not be redeemable until the regular delay (`LOCKED_FUNDS_PERIOD`) passes. If the validator desires to become a validator again, it would have to submit a new, separate bid.
+The new transaction would completely disable the validator's bid upon execution by the auction contract, which would set a flag on the bid, marking it as inactive, and remove the validator from the eras in which it had already won the auctions. The bid would remain in this state until the validator either bids again (possibly a zero amount, which would remove the "inactivity" flag and make it eligible for winning future auctions again) or explicitly retracts the bid.
```

Comment:
So that would change eras that had already been decided, and whose random seed is already known? Meaning that if you control N validators, you could choose one out of 2<sup>N</sup> leader sequences without even getting slashed, just by being inactive and getting your bids flagged.

I'd prefer to err on the safe side and only remove them from the validator set after the bonding period.

(We do currently choose the random bit only in the last block before the era… but even that block could be enough, if the attacker creates it: Then they can still choose from which of their validators to include a unit.)

Reply:
So basically wait for `AUCTION_DELAY` eras after the bid is disabled? Or was there another constant for the bonding period?

Reply:
You're right, that would be `AUCTION_DELAY`.